src/Microsoft.AspNet.WebSockets.Protocol/CommonWebSocket.cs

```diff
@@ -592,11 +592,11 @@ private void ValidateSegment(ArraySegment<byte> buffer)
             }
             if (buffer.Offset < 0 || buffer.Offset > buffer.Array.Length)
             {
-                throw new ArgumentOutOfRangeException(nameof(buffer.Offset), buffer.Offset, string.Empty);
+                throw new ArgumentOutOfRangeException(nameof(buffer), buffer.Offset, string.Empty);
```

Comment:
Shall we use this message **$"The {buffer.Offset} property cannot be negative and cannot be greater than the size of theArrayproperty's length."** instead